nautobot/dcim/tables/devices.py

```diff
@@ -975,3 +984,87 @@ class Meta(BaseTable.Meta):
         model = DeviceRedundancyGroup
         fields = ("pk", "name", "slug", "status", "failover_strategy", "member_count", "secrets_group", "tags")
         default_columns = ("pk", "name", "status", "failover_strategy", "member_count")
+
+
+#
+# Interface Redundancy Group
+#
+
+
+class InterfaceRedundancyGroupTable(BaseTable):
+    # pylint: disable=R0903
+    """Table for list view."""
+
+    pk = ToggleColumn()
+    name = tables.Column(linkify=True)
+    interfaces = tables.TemplateColumn(
+        template_code=INTERFACE_REDUNDANCY_GROUP_INTERFACES,
+        orderable=False,
+        verbose_name="Interfaces",
+    )
+    actions = ButtonsColumn(InterfaceRedundancyGroup)
+
+    class Meta(BaseTable.Meta):
+        """Meta attributes."""
+
+        model = InterfaceRedundancyGroup
+        fields = (
+            "pk",
+            "name",
+            "description",
+            "protocol",
+            "group_id",
+            "interfaces",
+        )
+
+
+class InterfaceRedundancyGroupAssociationTable(BaseTable):
+    # pylint: disable=R0903
+    """Table for list view."""
+
+    pk = ToggleColumn()
+    group = tables.Column(linkify=True, verbose_name="Group Name")
+    group__virtual_ip = tables.Column(linkify=True, verbose_name="Virtual IP")
+    group__group_id = tables.Column(verbose_name="Group ID")
+    priority = tables.TemplateColumn(template_code=INTERFACE_REDUNDANCY_INTERFACE_PRIORITY)
+    interface__device = tables.Column(linkify=True)
+    interface = tables.Column(linkify=True)
+    interface__status = tables.TemplateColumn(template_code=INTERFACE_REDUNDANCY_INTERFACE_STATUS)
+    group__status = tables.TemplateColumn(
+        template_code=INTERFACE_REDUNDANCY_GROUP_STATUS,
+        verbose_name="Group Status",
+    )
+    interface__ip_addresses = tables.TemplateColumn(
+        template_code=INTERFACE_IPADDRESSES,
+        orderable=False,
+        verbose_name="IP Addresses",
+    )
+    actions = ButtonsColumn(
+        model=InterfaceRedundancyGroupAssociation,
+        buttons=("edit", "delete"),
+    )
+
+    class Meta(BaseTable.Meta):
+        """Meta attributes."""
+
+        model = InterfaceRedundancyGroupAssociation
+        fields = (
+            "pk",
+            "group",
+            "interface",
+            "priority",
+            "group__status",
+            "group__virtual_ip",
+            "group__protocol",
+            "group__group_id",
+            "interface__device",
+            "interface__name",
+            "interface__status",
+            "interface__label",
+            "interface__enabled",
+            "interface__type",
+            "interface__description",
+            "interface__ip_addresses",
+        )
+
+        default_columns = ("priority", "actions")
```

Comment:
Are these two the only default columns we need here? probably "pk", "group", "interface" and etc as well right?

Reply:
This `InterfaceRedundancyGroupAssociation` model is a through model, and this table is displayed only in panels for the detail views for two models:
- `Interface`
- `InterfaceRedundancyGroup`

Each of these displays different columns, so the displayed columns are added in the view's `get_extra_context` method.